src/fluree/db/ledger/indexing.clj

```diff
@@ -107,216 +87,230 @@
-              (>! error-ch e)))
-      node)))
-
-(defn write-child-nodes
-  [conn network dbid idx error-ch parent child-nodes]
-  (let [cmp (:comparator parent)]
     (->> child-nodes
-         (map (fn [child]
-                (write-if-novel conn network dbid idx error-ch child)))
-         (async/map (fn [& written-nodes]
-                      (apply index/child-map cmp written-nodes))))))
-
-(defn write-descendants
-  "Writes the `descendants` of the branch node `parent`, adding new branch levels
-  if any branch node is too large"
-  [conn network dbid idx error-ch parent descendants]
-  (go-loop [children (<! (write-child-nodes conn network dbid idx error-ch parent descendants))]
-    (if (overflow-children? children)
-      (let [child-branches (rebalance-children parent children)]
-        (recur (<! (write-child-nodes conn network dbid idx error-ch parent child-branches))))
-      children)))
-
-(defn descendant?
-  [{:keys [rhs], first-flake :first, :as branch} node]
-  (if-not (index/branch? branch)
-    false
-    (let [cmp (:comparator branch)
-          {node-first :first, node-rhs :rhs} node]
-      (and (not (pos? (cmp first-flake node-first)))
-           (or (nil? rhs)
-               (and (not (nil? node-rhs))
-                    (not (pos? (cmp node-rhs rhs)))))))))
-
-(defn pop-descendants
-  "Pops all the descendants of `branch` off of the top of the stack `in-stack`"
-  [branch in-stack]
-  (loop [child-nodes []
-         stack       in-stack]
-    (let [nxt (peek stack)]
-      (if (and nxt (descendant? branch nxt))
-        (recur (conj child-nodes nxt) (pop stack))
-        [child-nodes stack]))))
-
-(defn write-tree
-  [conn network dbid idx error-ch node-stream]
-  (let [out (async/chan)]
-    (go-loop [stack []]
-      (if-let [node (<! node-stream)]
-        (if (index/leaf? node)
-          (recur (conj stack node))
-          (let [;; all descendants of a branch node should be at the top of the
-                ;; stack as long as the `node-stream` is in depth-first order
-                [descendants stack*] (pop-descendants node stack)
-                children    (<! (write-descendants conn network dbid idx error-ch
-                                                   node descendants))
-                first-flake (-> children first key)
-                branch      (-> node
-                                unmark-expanded
-                                (dissoc :id)
-                                (assoc :first first-flake
-                                       :children children))]
-            (recur (conj stack* branch))))
-        (async/pipe (write-if-novel conn network dbid idx error-ch (peek stack))
-                    out)))
-    out))
-
-(defn tally
-  [idx index-ch stat-ch]
+         (partition-all target-count)
+         (map (fn [kids]
+                (update-branch branch idx t kids)))
+         (fn [[maybe-leftmost & not-leftmost]]
+           (into [maybe-leftmost]
```

Comment:
I never knew about this arity for `into`! I think you may need to double-wrap this `fn` in parens in order for it to play nicely with the `->>`, though, or move it to a `defn`.